OpenRA.Game/FileFormats/IdxReader.cs

```diff
@@ -0,0 +1,74 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace OpenRA.FileFormats
+{
+	public class IdxReader
+	{
+		const string FileExtension = ".wav";
+
+		public readonly int SoundCount;
+		public IDictionary<string, IdxEntry> Entries;
+
+		public class IdxEntry
+		{
+			public int Offset;
+			public int Size;
+			public int SampleRate;
+			public int Flags;
+			public int Chunksize;
+		}
+
+		public IdxReader(Stream s)
+		{
+			s.Seek(0, SeekOrigin.Begin);
+
+			var id = s.ReadASCII(4);
+
+			if (id != "GABA")
+				throw new InvalidDataException(string.Format("Unable to load Idx file, did not find magic id, found {0} instead", id));
+
+			var two = s.ReadInt32();
+
+			if (two != 2)
+				throw new InvalidDataException(string.Format("Unable to load Idx file, did not find magic number 2, found {0} instead", two));
+
+			SoundCount = s.ReadInt32();
+
+			Entries = new Dictionary<string, IdxEntry>();
+
+			for (var i = 0; i < SoundCount; i++)
+				ReadNextEntry(s);
+		}
+
+		public void ReadNextEntry(Stream s)
+		{
+			var asciiname = s.ReadASCII(16);
+
+			int pos = asciiname.IndexOf('\0');
+			if (pos >= 0)
+				asciiname = asciiname.Substring(0, pos);
+
+			var entry = new IdxEntry
+			{
+				Offset = s.ReadInt32(),
+				Size = s.ReadInt32(),
+				SampleRate = s.ReadInt32(),
+				Flags = s.ReadInt32(),
+				Chunksize = s.ReadInt32()
+			};
+
+			Entries.Add(asciiname + FileExtension, entry);
```

Comment:
It does. But I don't know another way.

The index does not have extensions and contain flags such as bitrate, channels etc. The .bag binary blobs do not have this information.

* Without an extension but with header, the engine tries to load them as a RAW file which is wrong.
* Without header but with extension, the engine tries to load them and throws exceptions because it's missing the header info.

Do you maybe have a different solution?